src/auto/dbus_method_invocation.rs

```diff
@@ -0,0 +1,171 @@
+    }
+}
+
+impl DBusMethodInvocation {
+    pub fn get_connection(&self) -> Option<DBusConnection> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_connection(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_interface_name(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_interface_name(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_message(&self) -> Option<DBusMessage> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_message(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_method_info(&self) -> Option<DBusMethodInfo> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_method_info(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_method_name(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_method_name(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_object_path(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_object_path(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_parameters(&self) -> Option<glib::Variant> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_parameters(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_property_info(&self) -> Option<DBusPropertyInfo> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_property_info(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_sender(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_method_invocation_get_sender(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    //pub fn get_user_data(&self) -> /*Unimplemented*/Option<Fundamental: Pointer> {
+    //    unsafe { TODO: call gio_sys:g_dbus_method_invocation_get_user_data() }
+    //}
+
+    pub fn return_dbus_error(&self, error_name: &str, error_message: &str) {
+        unsafe {
+            gio_sys::g_dbus_method_invocation_return_dbus_error(
+                self.to_glib_full(),
+                error_name.to_glib_none().0,
+                error_message.to_glib_none().0,
+            );
+        }
+    }
+
+    //pub fn return_error(&self, domain: glib::Quark, code: i32, format: &str, : /*Unknown conversion*//*Unimplemented*/Fundamental: VarArgs) {
+    //    unsafe { TODO: call gio_sys:g_dbus_method_invocation_return_error() }
+    //}
+
+    pub fn return_error_literal(&self, domain: glib::Quark, code: i32, message: &str) {
```

Comment:
```suggestion
    pub fn return_error<T: glib::ErrorDomain>(&self, error: T, message: &str) {
```

Also if there are any other similar functions :)